### PR TITLE
Adds AuthDataResult to anonymous sign in

### PR DESCRIPTION
**Pull request by protocol86**
Adds `AuthDataResult` to the completion of anonymous sign as part of an effort to supplement all sign-in methods with `AdditionalUserInfo`.

### CODE REVIEW

File: Firebase/Auth/Source/FIRAuth.m [hunk 1]
```diff
@@ -676,6 +676,45 @@ - (void)signInWithCredential:(FIRAuthCredential *)credential
...
+    FIRAuthDataResultCallback decoratedCallback =
+        [self signInFlowAuthDataResultCallbackByDecoratingCallback:completion];
+    if (_currentUser.anonymous) {
+      FIRAdditionalUserInfo *additionalUserInfo =
+          [[FIRAdditionalUserInfo alloc] initWithProviderID:nil
+                                                    profile:nil
+                                                   username:nil
+                                                  isNewUser:NO];
+      decoratedCallback([[FIRAuthDataResult alloc] initWithUser:_currentUser
+                                             additionalUserInfo:additionalUserInfo],
```
Comment (ch40w31) [inline Firebase/Auth/Source/FIRAuth.m]: Extract the FIRAuthDataResult as a local var.

File: Firebase/Auth/Source/FIRAuth.m [hunk 2]
```diff
@@ -676,6 +676,45 @@ - (void)signInWithCredential:(FIRAuthCredential *)credential
...
+                             refreshToken:response.refreshToken
+                                anonymous:YES
+                                 callback:^(FIRUser *_Nullable user, NSError *_Nullable error) {
+        FIRAdditionalUserInfo *additionalUserInfo =
+          [[FIRAdditionalUserInfo alloc] initWithProviderID:nil
+                                                    profile:nil
+                                                   username:nil
+                                                  isNewUser:YES];
+        decoratedCallback([[FIRAuthDataResult alloc] initWithUser:user
+                                               additionalUserInfo:additionalUserInfo],
```
Comment (ch40w31) [inline Firebase/Auth/Source/FIRAuth.m]: same here

File: Firebase/Auth/Source/FIRAuth.m [hunk 3]
```diff
@@ -1092,6 +1128,23 @@ - (void)phoneNumberSignInWithRequest:(FIRVerifyPhoneNumberRequest *)request
...
+      [[FIRSignUpNewUserRequest alloc]initWithRequestConfiguration:_requestConfiguration];
+  [FIRAuthBackend signUpNewUser:request
+                       callback:^(FIRSignUpNewUserResponse *_Nullable response,
+                                  NSError *_Nullable error) {
+    if (error) {
+      completion(nil, error);
+      return;
+    }
+    completion(response, nil);
+  }];
```
Comment (ch40w31) [inline Firebase/Auth/Source/FIRAuth.m]: Looks like the callback block is equivalent of just calling `completion(response, error)`. If so, you can just pass `completion` to `FIRAuthBackend signUpNewUser:callback:` directly.

Comment (protocol86) [reply Firebase/Auth/Source/FIRAuth.m]: Great catch! done.

File: Firebase/Auth/Source/FIRAuth.m (hunk 2 above)
Comment (protocol86) [reply Firebase/Auth/Source/FIRAuth.m]: done.

File: Firebase/Auth/Source/FIRAuth.m (hunk 1 above)
Comment (protocol86) [reply Firebase/Auth/Source/FIRAuth.m]: I thought of doing this, it's certainly cleaner.

File: Example/Auth/Sample/MainViewController.m [hunk 4]
```diff
@@ -2786,6 +2794,24 @@ - (void)signInAnonymously {
...
+    @brief Signs in as an anonymous user, receiving an auth result containing a signed in user upon
+        success.
+ */
+- (void)signInAnonymouslyAuthDataResult {
+  [[AppManager auth] signInAnonymouslyAndRetrieveDataWithCompletion:
+      ^(FIRAuthDataResult *_Nullable authResult, NSError *_Nullable error) {
+    if (error) {
+      [self logFailure:@"sign-in anonymously failed" error:error];
+    }
+    else {
```
[the remark this answers is not in the document]
Comment (protocol86) [reply Example/Auth/Sample/MainViewController.m]: Done, thanks.

File: Firebase/Auth/Source/FIRAuth.m [hunk 5]
```diff
@@ -707,9 +708,10 @@ - (void)signInAnonymouslyAndRetrieveDataWithCompletion:(FIRAuthDataResultCallbac
                                                     profile:nil
                                                    username:nil
                                                   isNewUser:YES];
-        decoratedCallback([[FIRAuthDataResult alloc] initWithUser:user
-                                               additionalUserInfo:additionalUserInfo],
-                          nil);
+        FIRAuthDataResult *authDataResult =
+          [[FIRAuthDataResult alloc] initWithUser:user
```
Comment (ch40w31) [inline Firebase/Auth/Source/FIRAuth.m]: indent 4

**Pull request merged**